arrow-flight/src/sql/client.rs

```
@@ -227,6 +227,38 @@ impl FlightSqlServiceClient<Channel> {
         Ok(result.record_count)
     }
 
+    /// Execute a bulk ingest on the server and return the number of records added
+    pub async fn execute_ingest<S>(
+        &mut self,
+        command: CommandStatementIngest,
+        stream: S,
+    ) -> Result<i64, ArrowError>
+    where
+        S: Stream<Item = crate::error::Result<RecordBatch>> + Send + 'static,
+    {
+        let descriptor = FlightDescriptor::new_cmd(command.as_any().encode_to_vec());
+        let flight_data_encoder = FlightDataEncoderBuilder::new()
+            .with_flight_descriptor(Some(descriptor))
+            .build(stream);
+        // Safe unwrap, explicitly wrapped on line above.
+        let flight_data = flight_data_encoder.map(|fd| fd.unwrap());
```

This comment is no longer true, but is it still reasonable to unwrap or should we raise the error, or fuse the input stream, or...?

---

I think we should return the error from the client. Let me see if I can figure it out....